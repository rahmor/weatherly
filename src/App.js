import React from 'react';
import './App.css';

function App() {
  return (
    <main className='App'>
      <section className='App-daily'>
        <header className='App-daily-header'>
          <h1 className='App-title'>Weatherly</h1>
        </header>
        <form>
          <input type='search' placeholder='Search by City' />
          <button type='submit'>Search</button>
        </form>
        <h2>Atlanta, Ga</h2>
      </section>
    </main>
  );
}

export default App;
