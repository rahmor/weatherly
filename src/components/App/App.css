.App {
  display: flex;
  text-align: center;
  background: url('../../images/hexagon-2822770_1920.jpg') no-repeat
    center/cover;
  min-width: 100vw;
  min-height: 100vh;
  background-attachment: fixed;
}

.App-frame-frosted {
  min-height: 100%;
  min-width: 100%;
  background: inherit;
  position: relative;
  box-shadow: 0 0 1rem 0 rgba(0, 0, 0, 0.2);
  overflow: hidden;
  z-index: 1;
}

.App-frame-frosted:before {
  content: '';
  position: absolute;
  background: inherit;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  box-shadow: inset 0 0 2000px rgba(67, 50, 86, 0.5);
  filter: blur(10px);
  margin: -10px;
  z-index: -1;
}

.App-visually-hidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

@media screen and (min-width: 768px) {
  .App {
    padding: 50px;
  }
}
