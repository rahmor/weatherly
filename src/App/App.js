import React from 'react';
import Header from '../Header/Header';
import DailyWeather from '../DailyWeather/DailyWeather';
import './App.css';

function App() {
  return (
    <div className='App'>
      <section className='App-frame-frosted'>
        <Header />
        <main>
          <DailyWeather />
        </main>
      </section>
    </div>
  );
}

export default App;
