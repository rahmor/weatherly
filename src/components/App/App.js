import React from 'react';
import Header from '../Header/Header';
import CurrentWeather from '../CurrentWeather/CurrentWeather';
import lodash from 'lodash';
import { connect } from 'react-redux';
import { fetchWeather, updateCity } from '../../actions/actions';
import './App.css';

export class App extends React.Component {
  state = {
    CITIES: {
      'New York': [40.7127281, -74.0060152],
      Atlanta: [33.7490987, -84.3901849],
      'Los Angeles': [34.0536909, -118.2427666],
      Miami: [25.7741728, -80.19362],
      'San Francisco': [37.7790262, -122.4199061],
      Chicago: [41.8755616, -87.6244212],
      'Washington, DC': [38.8949855, -77.0365708],
      Boston: [42.3602534, -71.0582912],
    },
  };

  componentDidMount() {
    let cities = Object.keys(this.state.CITIES);
    let city = lodash.sample(cities);
    this.props.fetchWeather(this.state.CITIES[city]);
    this.props.updateCity(city);
  }

  render() {
    return (
      <div className='App'>
        <section className='App-frame-frosted'>
          <Header {...this.props} />
          <main>
            <CurrentWeather {...this.props} />
          </main>
        </section>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return state;
}

const mapDispatchToProps = {
  fetchWeather,
  updateCity,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
