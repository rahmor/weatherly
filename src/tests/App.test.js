import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('App Component', () => {
  it('should render', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
