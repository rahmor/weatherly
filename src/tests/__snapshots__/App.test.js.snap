// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<div
  className="App"
>
  <section
    className="App-frame-frosted"
  >
    <Header />
    <main>
      <DailyWeather />
    </main>
  </section>
</div>
`;
